content/Blog/2018-02-01-design-the-best-architecture-ever.md

```diff
@@ -0,0 +1,79 @@
+It is ultimately left up to the architects of the system to decide which structures to select and how many to implement.
+
+#### 5. Documenting the architecture
+
+1. **A document outlining the entire architecture** - 
+This step involves creating a table of contents that outlines the information and documentation available for the architecture, such as:
+- an overview of the embedded system
+- the actual requirements supported by the architecture
+- the definitions of the various structures
+- the inter-relationships between the structures outlining the documentation representing the various structures
+- how these documents are laid out (i.e., modelling techniques, notation, semantics, etc.), and so on.
+
+2. **A document for each structure**
+
+>This document should indicate which requirements are being supported by the structure and how these requirements are being supported by the design, as well as any relative constraints, issues, or open items.
+This document should also contain graphical and non-graphical (i.e., tabular, text, etc.) representations of each of the various elements within the structure. For instance, a graphical representation of the structural elements and relationships would include an index containing a textual summary of the various elements, their behavior, their interfaces, and their relationships to other structural elements.
+
+This document should be easily adaptable in specification formalism like UML (*Unified Modelling Language*).
+
+
+#### 6. Analyzing and evaluating the architecture
+
+>While there are many purposes for reviewing an architecture, primarily it is to determine if an architecture meets the requirements, and to evaluate the potential risks, and possible failure, of a design long before it has been built. When evaluating an architecture, both the person reviewing it and the process of the evaluation must be established.
+
+To compare and choose the best architecture between all proposals, we used grades to note each one in function of their behavior in multiple scenarios. It is called *qualitative-analysis approach*.
+
+Then we take into account the reviews from each module’s team. Indeed, they had to read and verify the documented architecture produced to ensure it matches the set of their own requirements. If one need is not answered, the architecture has to be redesigned by integrating the correction and follow the same process: it is a small V-Model.
+
+
+#### summary:
+This representation sums up all the steps from the creation to the maintenance of an architecture:
+{{<
+    image_pop_up_legend
+    "/images/Blog/lifecycle.png"
+    "Figure 1: Embedded Systems Design and Development Lifecycle Model"
+    "Figure 1: Embedded Systems Design and Development Lifecycle Model"
+>}}
+
+
+----------
+
+
+### SOURCES
+- *Embedded Systems Architectures* - chapitre 11 (ISBN: 978-0-7506-7792-9)
+[Link to ScholarVox](http://scholarvox.campuseiffel.fr/book/40000649)

```

Comment:
Hum... Isn't it possible to redirect to another public link instead of a link that is specific to "Campus Eiffel" ?